File: packages/sdk/src/mock.ts
import type { FlagValue, EdgeFlagsOptions } from './types.js';
import { EdgeFlags } from './client.js';

export interface MockOptions {
  flags?: Record<string, FlagValue>;
  configs?: Record<string, unknown>;
}

export function createMockClient(options: MockOptions = {}): EdgeFlags {
  const client = new EdgeFlags({
    token: 'mock_token',
    baseUrl: 'http://localhost',
    _mock: {
      flags: options.flags ?? {},
      configs: options.configs ?? {},
    },
  } as EdgeFlagsOptions & { _mock: { flags: Record<string, FlagValue>; configs: Record<string, unknown> } });

  return client;
}

File: packages/python/src/edgeflags/mock.py
from __future__ import annotations

from typing import Any

from .client import EdgeFlags, EdgeFlagsSync
from .types import FlagValue


def create_mock_client(
    flags: dict[str, FlagValue] | None = None,
    configs: dict[str, Any] | None = None,
) -> EdgeFlags:
    return EdgeFlags(
        token="mock_token",
        base_url="http://localhost",
        _mock={"flags": flags or {}, "configs": configs or {}},
    )


def create_mock_client_sync(
    flags: dict[str, FlagValue] | None = None,
    configs: dict[str, Any] | None = None,
) -> EdgeFlagsSync:
    return EdgeFlagsSync(
        token="mock_token",
        base_url="http://localhost",
        _mock={"flags": flags or {}, "configs": configs or {}},
    )
